### PR TITLE
[decoupled-execution] small cleanup and implement commit proof forward & reset

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

1. sync info now must carry highest ledger info
2. better structure the block store
3. add stub function in sync manager for followup PR
4. delete unused counters

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/diem/diem/tree/main/developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: consensus/src/block_storage/block_store.rs [hunk 1]
```diff
@@ -294,37 +258,29 @@ impl BlockStore {
...
-
-        // asynchronously execute and commit
+        // This callback is invoked synchronously withe coupled-execution and asynchronously in decoupled setup.
         self.state_computer
             .commit(
                 &blocks_to_commit,
                 finality_proof,
                 Box::new(
-                    move |executed_blocks: &[Arc<ExecutedBlock>],
+                    move |committed_blocks: &[Arc<ExecutedBlock>],
```
Comment (yuxiamit) [inline consensus/src/block_storage/block_store.rs:268]: We added some comments at the BufferManager that the callback should be valid if we merge the batches and only call the last callback. Maybe let's also mention it here.

File: consensus/consensus-types/src/sync_info.rs [hunk 2]
```diff
@@ -94,8 +94,8 @@ impl SyncInfo {
     ) -> Self {
         Self::new_decoupled(
             highest_quorum_cert,
-            highest_ordered_cert,
-            None,
+            highest_ordered_cert.clone(),
```
Comment (yuxiamit) [inline consensus/consensus-types/src/sync_info.rs:97]: looks like we can remove this clone by changing the order.

File: consensus/src/counters.rs [hunk 3]
```diff
@@ -272,67 +272,3 @@ pub static BLOCK_RETRIEVAL_CHANNEL_MSGS: Lazy<IntCounterVec> = Lazy::new(|| {
     )
     .unwrap()
 });
-
```
Comment (yuxiamit) [inline consensus/src/counters.rs:275]: I added a TODO item for the counters of buffer manager + phases.

**Review by yuxiamit — COMMENTED** [review]
looks good to me

**Review by msmouse — DISMISSED** [review]
This is a blind accept.

**Pull request closed without merging**